src/EPiServer.SocialAlloy.Web/Social/Tutorials/LikeButton/LikeButtonBlockController.cs

```diff
@@ -0,0 +1,150 @@
+                            Rater = Reference.Create(raterUser),
+                            Targets = new List<Reference> { Reference.Create(targetPage) }
+                        },
+                        PageInfo = new PageInfo() { PageSize = 1 }
+                    }
+                );
+
+                // Add the current Like rating, if any, to the block view model. If the user is logged 
+                // into the site and had previously liked the current page then the CurrentRating value
+                // should be 1 (LIKED_RATING).  Anonymous user Likes are generated with unique random users
+                // and thus the current anonymous user will never see a current rating value as he/she
+                // can Like the page indefinitely.
+                if (ratingPage.Results.Count() > 0)
+                {
+                    blockModel.CurrentRating = ratingPage.Results.ToList().FirstOrDefault().Value.Value;
+                }
+
+                // Using the EPiServer Social Rating service, get the existing Like statistics for the page (target)
+                var ratingStatisticsPage = ratingService.Get(
+                    new Criteria<RatingStatisticsFilter>()
+                    {
+                        Filter = new RatingStatisticsFilter()
+                        {
+                            Targets = new List<Reference> { Reference.Create(targetPage) }
+                        },
+                        PageInfo = new PageInfo() { PageSize = 1 }
+                    }
+                );
+
+                // Add the page Like statistics to the block view model
+                if (ratingStatisticsPage.Results.Count() > 0)
+                {
+                    var statistics = ratingStatisticsPage.Results.ToList().FirstOrDefault();
+                    if (statistics.TotalCount > 0)
+                    {
+                        blockModel.TotalCount = statistics.TotalCount;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // the rating service may throw a number of possible exceptions
+                // should handle each one accordingly -- see rating service documentation
+            }
+
+            return PartialView("~/Views/Social/LikeButtonBlock/LikeButtonView.cshtml", blockModel);
+        }
+
+        /// <summary>
+        /// Submit handles a click of the Like button.  It accepts a Like button block model,
+        /// saves the Like rating, and redirects back to the current page.
+        /// </summary>
+        /// <param name="likeButtonBlock">The Like button block model.</param>
+        /// <returns>Result of the redirect to the current page.</returns>
+        [HttpPost]
+        public ActionResult Submit(LikeButtonBlockViewModel likeButtonBlock)
+        {
+            var pageLink = this.pageRouteHelper.PageLink;
+
+            var targetPage = this.pageRouteHelper.Page.ContentGuid.ToString();
```

Comment:
Do you always get the right page back here? @BenSchaefer found, in the work that he's doing now, that this didn't always return the expected page on post back.